modules/wallet/transactionbuilder.go

```diff
@@ -9,6 +9,90 @@ import (
 	"github.com/NebulousLabs/Sia/types"
 )
 
+// walletSpender allows clients to retrieve and modify keys and outputs in a
+// wallet in order to spend siacoins.
+type walletSpender interface {
+	// acquireLock acquires a mutex on walletSpender. Clients must call
+	// acquireLock before calling any other function in walletSpender and must
+	// call releaseLock before exiting scope.
+	acquireLock()
+	releaseLock()
+
+	siacoinOutputs() map[types.SiacoinOutputID]types.SiacoinOutput
+	siafundOutputs() map[types.SiafundOutputID]types.SiafundOutput
+	unconfirmedProcessedTransactions() []modules.ProcessedTransaction
+
+	// spendHeight returns the block height at which a given output was spent.
+	spendHeight(types.OutputID) types.BlockHeight
+
+	// setSpendHeight marks the output with the given ID as spent in the given
+	// block.
+	setSpendHeight(types.OutputID, types.BlockHeight)
+
+	// key returns the key associated with the given hash.
+	key(types.UnlockHash) spendableKey
+
+	// hasKey indicates whether a key with the given hash exists in the wallet.
+	hasKey(types.UnlockHash) bool
+
+	// consensusSetHeight returns the current block height of the wallet.
+	consensusSetHeight() types.BlockHeight
+
+	// nextPrimarySeedAddress fetches the next address from the primary seed.
+	nextPrimarySeedAddress() (types.UnlockConditions, error)
+}
+
+// spendableWallet is a wrapper over Wallet that implements walletSpender.
+type spendableWallet struct {
+	wallet *Wallet
+	lockID int
+}
```

Comment:
Any interface that you create to use with the transaction builder should be satisfied by the wallet directly, instead of using a secondary struct that has a wallet inside of it.